client/lib/plans/index.js

```diff
@@ -67,7 +67,33 @@ export function addCurrentPlanToCartAndRedirect( sitePlans, selectedSite ) {
 }
 
 export function getCurrentPlan( plans ) {
-	return find( plans, { currentPlan: true } );
+	const currentPlan = find( plans, { currentPlan: true } );
+
+	if ( currentPlan ) {
+		return currentPlan;
+	}
+
+	// the site doesn't have defined a current plan.
+	// Let's dive in its `plan` property
+	const site = sitesList.getSelectedSite();
+
+	if ( ! site ) {
+		return null;
+	}
+
+	const { plan } = site;
+
+	if ( ! plan ) {
+		return null;
+	}
+
+	// let's check if the current plan is the old `Enterprise` plan.
+	// if it is then let's assing `business` plan.
+	if ( plan.product_id === 1007 ) {
```

Comment:
we could add `isEnterprise()` function but I'm afraid that finally we'll ending checking of the same `1007` code. Let me implement something though.